src/main/kotlin/io/github/vootelerotov/renovator/Renovator.kt

```diff
@@ -81,48 +80,90 @@ class Renovator : CliktCommand() {
 
   private fun merge(pr: PullRequest, pullRequestClient: PullRequestClient, confirmedViaFlag: Boolean = false) {
     if (pr.merged() != false) {
-      echo("PR ${prDescription(pr)} is already merged")
+      echo("PR with ${prDescriptionShort(pr)} is already merged")
       return
     }
 
     if (pr.mergeable().getOrNull() != true) {
-      echo("PR ${prDescription(pr)} cannot be merged")
+      echo("PR with ${prDescriptionShort(pr)} cannot be merged")
       return
     }
 
-    val isConfirmed = confirmedViaFlag || confirm("Approve and merge PR ${prDescription(pr)}?")
-      ?: throw IllegalStateException("Can not get confirmation from stdin")
+    val isConfirmed = confirmedViaFlag || confirmMerge(pr)
 
     if (!isConfirmed) {
       return
     }
 
-    val comment = prompt("Enter comment to approve the PR with", default = defaultComment)
+    val comment = if (defaultComment.isNullOrBlank()) {
+      prompt("Enter comment to approve the PR with")
+    } else {
+      prompt(
+        "Press enter to approve the PR with the default comment '$defaultComment' or enter a different comment to approve PR with",
+        default = defaultComment,
+        showDefault = false
+      )
+    }
       ?: throw IllegalStateException("Can not get comment from stdin")
 
-    echo("Approving PR ${prDescription(pr)} with comment $comment")
+    echo("Approving PR ${pr.title()} with comment $comment")
 
     pullRequestClient.createReview(pr.number()!!, ImmutableReviewParameters.builder()
       .body(comment)
       .event(APPROVE_EVENT)
       .build()
     ).get()
-    echo("Merging PR ${prDescription(pr)} via re-base")
+    echo("Merging PR ${pr.title()} via re-base")
     pullRequestClient.merge(
       pr.number()!!,
       ImmutableMergeParameters.builder().mergeMethod(MergeMethod.rebase).sha(pr.head()?.sha()!!).build()
     ).get()
     echo("")
   }
 
+  private fun confirmMerge(pr: PullRequest): Boolean {
+    val input = prompt(
+      "Approve and merge following PR [y/N]? Type 'i' to get more information about the PR. ${prDescriptionLong(pr)}",
+      default = "N",
+      showDefault = false
+    )
+
+    return if ("i".equals(input, ignoreCase = true)) {
+      echo("""
+        
+        Title: ${pr.title()}
+        URL: ${pr.url()}
+        
+      """.trimIndent())
+      confirmMerge(pr)
```

Comment:
Eloquent way of handling this with recursion, a fan -- I was thinking that we do not want to keep state and like this, we do not have to. Props.